src/hooks/useAuth.tsx

```diff
@@ -12,46 +12,45 @@ const key = (str: string) => `src/hooks/useAuth/${str}`
 const idTokenState = atom<string>({
   key: key('idTokenState'),
   default: '',
-})
-
-export const useAuth = () => {
-  const [idToken, setIdToken] = useRecoilState(idTokenState)
-
-  const isSignedIn = useMemo(() => !!idToken, [idToken])
+  effects: [
+    ({ setSelf }) => {
```

Comment:
Use [Atom Effects](https://recoiljs.org/docs/guides/atom-effects/) to observe a firebase auth status.